jumanji/training/networks/connector/actor_critic.py

```diff
@@ -138,7 +132,11 @@ def __init__(
         self.env_time_limit = env_time_limit
 
     def __call__(self, observation: Observation) -> chex.Array:
-        grid = jax.vmap(process_grid)(observation.grid)  # (B, N, G, G, 2)
+        agent_ids = jnp.arange(self.num_agents)
+        # (N, B, G, G, 2)
+        grid = jax.vmap(process_grid, (None, 0))(observation.grid, agent_ids)
+        grid = jnp.transpose(grid, (1, 0, 2, 3, 4))  # (B, N, G, G, 2)
```

Comment:
Alternatively, you can use `out_axes` and then no need to transpose.
```suggestion
        grid = jax.vmap(process_grid, in_axes=(None, 0), out_axes=1)(observation.grid, agent_ids)  # (B, N, G, G, 2)
```